### ISSUE TITLE
Remove `flush()` from `Channel`

### ISSUE THREAD

**Issue by tindzk** [MetaStack-pl/MetaRx#35]
`flush()` should be removed from `Channel`. See also [this discussion](https://github.com/MetaStack-pl/MetaRx/commit/a05c11828401504d8524baf99dbc8e4bad94d93b#commitcomment-13466442).

**Comment by mkotsbak** [comment]
Where exactly in the code do you see flush()? In ReadChannel? Be sure not to break zip() with ReadChannels derived from Vars (tests should break in that case).

**Comment by tindzk** [comment]
Yes, `flush()` is defined [here](https://github.com/MetaStack-pl/MetaRx/blob/master/shared/src/main/scala/pl/metastack/metarx/Channel.scala#L71).

**Comment by mkotsbak** [comment]
Ah ok, it is created for new ReadChannels. But I guess it is needed for zip to work.

**Comment by tindzk** [comment]
There could be two implementations of `zip()`: It is defined in `ReadChannel` and then overridden in `ReadStateChannel` (where we implement the `flush()` function on the resulting channel). I would also change the signature to take a `ReadStateChannel` in `zip()`, so that we don't need to duplicate the state internally.

**Comment by mkotsbak** [comment]
Japp, that is a possibility. There are other uses of flush as well to think of:
bind() must be moved to StateChannel?

I also think ReadStateChannel should override map() to return a new ReadStateChannel, so that it keeps the flush() and is usable in zip().
